test/integration/node-specific/ipv6.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from 'chai';
+import * as net from 'net';
+import * as process from 'process';
+import * as sinon from 'sinon';
+
+import { ConnectionCreatedEvent, MongoClient, ReadPreference, TopologyType } from '../../../src';
+import { byStrings, sorted } from '../../tools/utils';
+
+describe('IPv6 Addresses', () => {
+  let client: MongoClient;
+  let ipv6Hosts: string[];
+
+  beforeEach(async function () {
+    if (
+      process.platform !== 'win32' ||
+      this.configuration.topologyType !== TopologyType.ReplicaSetWithPrimary
+    ) {
+      if (this.currentTest) {
+        // Ubuntu 18 does not support localhost AAAA lookups (IPv6)
+        // Windows (VS2019) has the AAAA lookup
+        // We do not run a replica set on macos
+        this.currentTest.skipReason =
+          'We are only running this on windows currently because it has the IPv6 translation for localhost';
+      }
+      return this.skip();
+    }
+
+    ipv6Hosts = this.configuration.options.hostAddresses.map(({ port }) => `[::1]:${port}`);
+    client = this.configuration.newClient(`mongodb://${ipv6Hosts.join(',')}/test`, {
+      [Symbol.for('@@mdb.skipPingOnConnect')]: true,
+      maxPoolSize: 1
+    });
+  });
+
+  afterEach(async function () {
+    sinon.restore();
+    await client?.close();
+  });
+
+  it('should have IPv6 loopback addresses set on the client', function () {
+    const ipv6LocalhostAddresses = this.configuration.options.hostAddresses.map(({ port }) => ({
+      host: '::1',
+      port,
+      isIPv6: true,
+      socketPath: undefined
+    }));
+    expect(client.options.hosts).to.deep.equal(ipv6LocalhostAddresses);
+  });
+
+  it('should successfully connect using IPv6 loopback addresses', async function () {
+    const localhostHosts = this.configuration.options.hostAddresses.map(
+      ({ port }) => `localhost:${port}` // ::1 will be swapped out for localhost
+    );
+    await client.db().command({ ping: 1 });
+    // After running the first command we should receive the hosts back as reported by the mongod in a hello response
+    // mongodb will report the bound host address, in this case "localhost"
+    expect(client.topology).to.exist;
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
is it time to add the no-non-null-assertion to the test ts config? (not in this PR, necessarily, but it's going to be an issue for all tests where we assert existence via `expect`)

Reply:
Yea we can do that, if we add it the linter should remove all the execeptions we have added (this included) so I can put that up seperately